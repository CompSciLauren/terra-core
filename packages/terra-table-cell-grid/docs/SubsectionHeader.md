# Terra Table Subsection Header

The Terra Subsection Header is built upon a table row and can be placed similarly within a table, provided the number of columns are provided to the SubsectionHeader. The subsection header should only be placed within the table via children when the header is static, as the actual row collapse is handled by the section.

## Usage

```jsx
import React from 'react';
import { SubsectionHeader } from 'terra-flex-table';

<SubsectionHeader
  key="unique-react-key"
  title="Subsection Title"
/>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
