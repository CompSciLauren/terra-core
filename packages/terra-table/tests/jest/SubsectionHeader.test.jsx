import React from 'react';
import SubsectionHeader from '../../src/SubsectionHeader';

// Snapshot test
it('should render a default subsection header', () => {
  const shallowComponent = shallow(
    <SubsectionHeader title="test" />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a subsection header with isCollpased', () => {
  const shallowComponent = shallow(
    <SubsectionHeader title="test" isCollapsed />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a subsection header with isCollapsible', () => {
  const shallowComponent = shallow(
    <SubsectionHeader title="test" isCollapsible />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a subsection header with level', () => {
  const shallowComponent = shallow(
    <SubsectionHeader title="test" level={4} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a subsection header with metaData', () => {
  const shallowComponent = shallow(
    <SubsectionHeader title="test" metaData={{}} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a subsection header with onSelect', () => {
  const mockCallBack = jest.fn();

  const shallowComponent = shallow(
    <SubsectionHeader title="test" isCollapsible onSelect={mockCallBack} />,
  );
  expect(shallowComponent).toMatchSnapshot();
  shallowComponent.find('.subsection-header').simulate('click');
  shallowComponent.find('.subsection-header').simulate('keydown', { nativeEvent: { keyCode: 13 } });
  shallowComponent.find('.subsection-header').simulate('keydown', { nativeEvent: { keyCode: 32 } });
  expect(mockCallBack.mock.calls.length).toEqual(3);
});

it('should render a subsection header with refCallback', () => {
  const shallowComponent = shallow(
    <SubsectionHeader title="test" refCallback={jest.fn()} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});
