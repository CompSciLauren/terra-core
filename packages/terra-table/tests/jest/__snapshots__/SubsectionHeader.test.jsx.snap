// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default subsection header 1`] = `
<div
  className="subsection-header"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <h3
      className="title"
    >
      test
    </h3>
  </div>
</div>
`;

exports[`should render a subsection header with isCollapsible 1`] = `
<div
  aria-expanded={true}
  className="subsection-header is-collapsible"
  data-row-show-focus="true"
  onBlur={[Function]}
  onMouseDown={[Function]}
  tabIndex="0"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <div
      className="start"
    >
      <span
        className="accordion-icon is-open"
      />
    </div>
    <div
      className="fill"
    >
      <h3
        className="title"
      >
        test
      </h3>
    </div>
  </div>
</div>
`;

exports[`should render a subsection header with isCollpased 1`] = `
<div
  className="subsection-header"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <h3
      className="title"
    >
      test
    </h3>
  </div>
</div>
`;

exports[`should render a subsection header with level 1`] = `
<div
  className="subsection-header"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <h4
      className="title"
    >
      test
    </h4>
  </div>
</div>
`;

exports[`should render a subsection header with metaData 1`] = `
<div
  className="subsection-header"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <h3
      className="title"
    >
      test
    </h3>
  </div>
</div>
`;

exports[`should render a subsection header with onSelect 1`] = `
<div
  aria-expanded={true}
  className="subsection-header is-collapsible"
  data-row-show-focus="true"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  tabIndex="0"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <div
      className="start"
    >
      <span
        className="accordion-icon is-open"
      />
    </div>
    <div
      className="fill"
    >
      <h3
        className="title"
      >
        test
      </h3>
    </div>
  </div>
</div>
`;

exports[`should render a subsection header with refCallback 1`] = `
<div
  className="subsection-header"
>
  <div
    aria-level="2"
    className="subsection-content"
    role="heading"
  >
    <h3
      className="title"
    >
      test
    </h3>
  </div>
</div>
`;
